### PR TITLE
Feature/662 cleanup docker setup

### PR DESCRIPTION
**Pull request by kcinay055679**

### CODE REVIEW

**Review by Robin481 — CHANGES_REQUESTED** [review]
I have no Idea how to test this.
And as other people might want to use this application with a dockerized setup please update the README and explain the user how to use it. Thanks!

**Review by Robin481 — CHANGES_REQUESTED** [review]
Update Nodesource script and fix `devise_mapping` login bug

File: lib/auth_config.rb [hunk 1]
```diff
@@ -24,20 +26,31 @@ def admin_role
...
+        ENV.fetch(key.to_s.upcase, default)
       end
     end
 
     def local?
-      ENV.fetch('LOCAL', false)
+      to_boolean(ENV.fetch('LOCAL', false))
+    end
+
+    def to_boolean(value)
```
Comment (Robin481) [inline lib/auth_config.rb:49]: already exists: https://github.com/puzzle/skills/blob/master/app/controllers/concerns/param_converters.rb#L7

**Review by Robin481 — CHANGES_REQUESTED** [review]

File: .github/workflows/rails.yml [hunk 2]
```diff
@@ -39,19 +39,19 @@ jobs:
       - name: 'Run database migrations'
         env:
           RAILS_ENV: test
-          PGHOST: ${{ env.PGHOST }}
-          POSTGRES_DB: ${{ env.POSTGRES_DB }}
-          POSTGRES_USER: ${{ env.POSTGRES_USER }}
-          POSTGRES_PASSWORD: ${{ env.POSTGRES_PASSWORD }}
+          RAILS_DB_HOST: ${{ env.PGHOST }}
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply .github/workflows/rails.yml:42]: as we discussed

File: config/docker/development/Rails.dockerfile [hunk 3]
```diff
@@ -9,12 +9,11 @@ WORKDIR /myapp
...
 COPY ./rails-entrypoint /usr/local/bin
 
-RUN curl -sL https://deb.nodesource.com/setup_14.x | bash -
+RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash -
 RUN apt-get update
 RUN apt-get install direnv -y
 RUN apt-get install firefox-esr -y
-RUN apt-get install nodejs -y
-RUN apt-get install npm -y
+RUN apt-get install -y nodejs
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply config/docker/development/Rails.dockerfile:16]: No we still need to install nodejs via apt according to the [docs](https://github.com/nodesource/distributions?tab=readme-ov-file#using-ubuntu-2)

File: app/helpers/auth_helper.rb [hunk 4]
```diff
@@ -9,7 +9,16 @@ def admin?
     current_auth_user&.is_admin
   end
 
+  def find_person_by_auth_user
+    Person.find_by(name: current_auth_user&.name)
+  end
+
   def development?
-    ENV['DEVELOPMENT'] == 'true' && ENV['RAILS_ENV'] == 'development'
+    ENV['DEVELOPMENT'] == 'true' && Rails.env.development?
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/helpers/auth_helper.rb]: we do, at least now

File: docker-compose.yml [hunk 5]
```diff
@@ -1,54 +1,56 @@
...
+  assets:
+    container_name: assets
+    image: ruby:3.2
+    working_dir: /myapp
+    command: >
+     /bin/bash -c "
+      apt-get update &&
+      apt-get install nodejs npm -y &&
+      npm install -g yarn && bin/assets &&
+      sleep infinity"
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply docker-compose.yml:43]: no

**Pull request merged**